File: app/src/main/java/com/remotearthsolutions/expensetracker/adapters/CategoryListAdapter.java
package com.remotearthsolutions.expensetracker.adapters;


import android.app.Activity;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.remotearthsolutions.expensetracker.R;
import com.remotearthsolutions.expensetracker.adapters.viewholder.CategoryViewHolder;
import com.remotearthsolutions.expensetracker.databaseutils.models.CategoryModel;

import java.util.List;

public class CategoryListAdapter extends RecyclerView.Adapter<CategoryViewHolder> {

    private List<CategoryModel> categorylist;
    private CategoryListAdapter.OnItemClickListener listener;
    private String selectedItem;

    public CategoryListAdapter(List<CategoryModel> categorylist, String selectedItem) {
        this.categorylist = categorylist;
        this.selectedItem = selectedItem;
    }

    public CategoryListAdapter(List<CategoryModel> categorylist) {
        this.categorylist = categorylist;

    }

    public void setOnItemClickListener(CategoryListAdapter.OnItemClickListener listener) {
        this.listener = listener;
    }

    @NonNull
    @Override
    public CategoryViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.custom_catagory, parent, false);
        return new CategoryViewHolder(view, listener);
    }

    @Override
    public void onBindViewHolder(@NonNull CategoryViewHolder holder, int position) {
        CategoryModel category = categorylist.get(position);
//        if (category.getName().equals(bundle.getString("name"))) {
//            Log.d("Hellloooo", "onBindViewHolder: "+ categorylist.get(position));
//        }
        holder.bind(category);
    }

    @Override
    public int getItemCount() {
        if (categorylist == null) return 0;
        return categorylist.size();
    }

    public interface OnItemClickListener {
        void onItemClick(CategoryModel category);
    }
}

File: app/src/main/java/com/remotearthsolutions/expensetracker/adapters/CategoryListAdapter.kt
package com.remotearthsolutions.expensetracker.adapters

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.remotearthsolutions.expensetracker.R
import com.remotearthsolutions.expensetracker.adapters.viewholder.CategoryViewHolder
import com.remotearthsolutions.expensetracker.databaseutils.models.CategoryModel
import com.remotearthsolutions.expensetracker.utils.Utils.ScreenSize

class CategoryListAdapter : RecyclerView.Adapter<CategoryViewHolder> {
    private var categorylist: List<CategoryModel>?
    private var listener: OnItemClickListener? = null
    private var selectedCategoryId = 0
    private var screenSize: ScreenSize? = null

    constructor(categorylist: List<CategoryModel>?, selectedCategoryId: Int) {
        this.categorylist = categorylist
        this.selectedCategoryId = selectedCategoryId
    }

    constructor(categorylist: List<CategoryModel>?) {
        this.categorylist = categorylist
    }

    fun setOnItemClickListener(listener: OnItemClickListener?) {
        this.listener = listener
    }

    fun setScreenSize(screenSize: ScreenSize?) {
        this.screenSize = screenSize
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CategoryViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.custom_catagory, parent, false)
        return CategoryViewHolder(view, listener!!, screenSize)
    }

    override fun onBindViewHolder(holder: CategoryViewHolder, position: Int) {
        val category = categorylist!![position]
        holder.bind(category, category.id == selectedCategoryId)
    }

    override fun getItemCount(): Int {
        return if (categorylist == null) {
            0
        } else categorylist!!.size
    }

    interface OnItemClickListener {
        fun onItemClick(category: CategoryModel?)
    }
}